### PR TITLE
update storageclient to reflect as a clusterscoped resource

### PR DESCRIPTION
**Pull request by leelavg**
Storageclient is now a clusterscoped resource and this PR reflects the changes for it to work as such.

Depends on red-hat-storage/ocs-client-operator#133

### CODE REVIEW

File: controllers/storagecluster/storageclient.go [hunk 1]
```diff
@@ -30,11 +28,10 @@ func (s *storageClient) ensureCreated(r *StorageClusterReconciler, storagecluste
 	}
 
 	storageClient := &ocsclientv1a1.StorageClient{}
-	storageClient.Name = storagecluster.Name + "-client"
-	storageClient.Namespace = r.OperatorNamespace
+	storageClient.Name = storagecluster.Name
 	_, err := cutil.CreateOrUpdate(r.ctx, r.Client, storageClient, func() error {
-		if err := controllerutil.SetControllerReference(storagecluster, storageClient, r.Scheme); err != nil {
```
Comment (nb-ohad) [inline controllers/storagecluster/storageclient.go:36]: `controllerutil` is the more common name used in many repos. Can we please align on this and not `cutils`?

**Review by nb-ohad — CHANGES_REQUESTED** [review]

File: controllers/storagecluster/storageclient.go (hunk 1 above)
Comment (leelavg) [reply controllers/storagecluster/storageclient.go:36]: done.

**Comment by nb-ohad** [comment]
LGTM

**Comment by leelavg** [comment]
rebased.

**Pull request merged**